src/controllers/consumer.ts

```diff
@@ -68,6 +68,17 @@ export const getPublishedDatasetView = async (req: Request, res: Response) => {
   res.json(preview);
 };
 
+export const getPublishedDatasetFilters = async (req: Request, res: Response) => {
+  const dataset = await PublishedDatasetRepository.getById(res.locals.datasetId, withAll);
```

Comment:
It looks like

```
const filters = await getFilters(dataset.publishedRevision, lang);
```

only uses the revision id, so the `withAll` is an expensive hydration that we then don't use any of other than the revision id?

```suggestion
  const dataset = await PublishedDatasetRepository.getById(res.locals.datasetId, {
    publishedRevision: true
  });
```

or if you modify `getFilters` to just take an id instead of a revision object, you could just do `getFilters(dataset.publishedRevisionId, lang);` and not fetch any relations at all.